src/serialization/toSerializableFormat.js

```diff
@@ -52,13 +49,16 @@ function transformSubstate(originalSubState) {
  * @returns {*}
  */
 export function toSerializableFormat(state) {
-  const accumulator = _.isArray(state) ? [] : {};
+  if (!_.isObjectLike(state)) {
+    return state;
+  }
 
-  return _.reduce(state, (serializableState, substate, subStateKey) => {
-    const serializableSubState = transformSubstate(substate);
+  const transformedState = transformSubstate(state);
```

Comment:
I think we should change things a bit differently. It seems to me like `transformSubstate` should be the entry point for serialization and the reducer logic should be placed into helper which `tranformSubstate` uses. 

It seems to me like we are processing data twice in some cases.